estate-services/src/main/java/org/egov/ps/service/ApplicationService.java

```diff
@@ -172,6 +196,101 @@ public void updatePostPayment(ApplicationRequest applicationRequest, Map<String,
 			producer.push(config.getUpdateApplicationTopic(),
 					new ApplicationRequest(requestInfo, applicationsForUpdate));
 		}
+
 	}
 
+	private void postApprovalChangeOwnerShare(ApplicationRequest applicationRequest) {
+		/**
+		 * Change share % after OWNERSHIP_TRANSFER and update property topic
+		 */
+
+		RequestInfo requestInfo = applicationRequest.getRequestInfo();
+		for (Application application : applicationRequest.getApplications()) {
+			if (application.getModuleType().equalsIgnoreCase(PSConstants.OWNERSHIP_TRANSFER)) {
+
+				Owner newOwner = new Owner();
+//				Property property = propertyRepository.findPropertyById(application.getProperty().getId());
+
+				PropertyCriteria propertySearchCriteria = PropertyCriteria.builder()
+						.propertyId(application.getProperty().getId()).build();
+				List<Property> properties = propertyRepository.getProperties(propertySearchCriteria);
+
+//				properties.forEach(property -> {
+
+				Property property = properties.get(0);
```

Comment:
why did we replace forEach with properties.get(0). What do we do if there are no properties that are returned ?

Reply:
resolved